src/Qt-RSA/qrsaencryption.cpp

```diff
@@ -312,79 +307,88 @@ QRSAEncryption::QRSAEncryption() {
 QByteArray QRSAEncryption::encode(const QByteArray &rawData, const QByteArray &pubKey) {
 
     switch (pubKey.size()) {
-    case RSA_64 / 4: {
-        return encodeArray<uint64_t>(rawData, pubKey);
-    }
+        case RSA_64 / 4: {
+            return encodeArray<uint64_t>(rawData, pubKey);
+        }
 
-    case RSA_128 / 4: {
-        return encodeArray<uint128_t>(rawData, pubKey);
-    }
+        case RSA_128 / 4: {
+            return encodeArray<uint128_t>(rawData, pubKey);
+        }
 
-    default: return QByteArray();
+        default: return QByteArray();
     }
 }
 
-
 QByteArray QRSAEncryption::decode(const QByteArray &rawData, const QByteArray &privKey) {
 
     switch (privKey.size()) {
-    case RSA_64 / 4: {
-        return decodeArray<uint64_t>(rawData, privKey);
-    }
 
-    case RSA_128 / 4: {
-        return decodeArray<uint128_t>(rawData, privKey);
-    }
+        case RSA_64 / 4: {
+            return decodeArray<uint64_t>(rawData, privKey);
+        }
+
+        case RSA_128 / 4: {
+            return decodeArray<uint128_t>(rawData, privKey);
+        }
 
-    default: return QByteArray();
+        default: return QByteArray();
     }
 }
 
+// EDS (digital signature)
 QByteArray QRSAEncryption::signMessage(QByteArray rawData, const QByteArray &privKey) {
-    auto msg = encode(rawData, privKey);
-    int size = rawData.size();
-    rawData.insert(0, reinterpret_cast<char*>(&size), sizeof (int));
-    rawData.append(msg);
+
+    QByteArray hash = QCryptographicHash::hash(rawData, QCryptographicHash::Sha256);
+
+    QByteArray signature = encode(hash, privKey);
+
+    rawData.append(SIGN_MARKER + signature.toHex() + SIGN_MARKER);
+
+    qDebug() << hash;
 
     return rawData;
 }
 
+// check valid EDS
 bool QRSAEncryption::checkSignMessage(const QByteArray &rawData, const QByteArray &pubKey) {
-    int mSize = 0;
-    memcpy(&mSize, rawData.left(sizeof (int)), sizeof (int));
 
-    auto message = rawData.mid(sizeof (int), mSize);
-    auto sig = rawData.mid(mSize + static_cast<int>(sizeof (int)));
+    auto signStartPos = rawData.lastIndexOf(SIGN_MARKER, rawData.length() - QString(SIGN_MARKER).length() - 1),
+         signLength   = rawData.length() - signStartPos - QString(SIGN_MARKER).length() * 2;
 
-    return message == decode(sig, pubKey);
+    // message, that was recieved from channel
+    QByteArray message = rawData.left(signStartPos);
+
+    // signature, that was recieved and decrypt from channel
+    QByteArray signature = decode(QByteArray::fromHex(rawData.mid(signStartPos + QString(SIGN_MARKER).length(), signLength)),
+                                  pubKey);
+
+    // if decrypt signature == sha256(recived message), then signed message is valid
+    return signature == QCryptographicHash::hash(message, QCryptographicHash::Sha256);
 }
```

Comment:
You in one function do a call of function QString (SIGN_MARKER) .length ()
more than 1 time, you need to make the value of this divination in a separate variable.
since here is a call to the QString constructor and then a call to length, the variable will save a lot of time.

Reply:
change name of variable "signature" becouse it is contains hash of message but not signature.

Reply:
And dont use long strings 
Please transfer to a new line very long lines.
 
![image](https://user-images.githubusercontent.com/12465465/59275320-c4ed1200-8c64-11e9-9e71-c58a0b85d882.png)